provide resolution notes for channel slug mismatch error in 1.113.0

brianrhea opened the pull request:
Adding more detail about the channel mismatch errors beginning in 113

brianrhea commented:
@ajp-io @paigecalvert @AmberAlston @xavpaice Adding some detail to the 113 release notes to help vendors resolve this error. I'd love your feedback and suggestions on making sure this is accurate and very clear to vendors if they will be affected by this change.

@pandemicsyn is also going to [emit a more helpful message](https://replicated.slack.com/archives/CEEFDQURM/p1724251740888769?thread_ts=1724228929.171579&cid=CEEFDQURM) when we encounter a license validation error.

This isn't a "Known Issue" because it isn't going to fix itself in a future version. The install commands have to match the channel for which they are intended and that are displayed in the Vendor Portal and the CLI.

Open to suggestions and grateful for any improvements!

AmberAlston commented:
@brianrhea Can I still have my install command be `kots install app` or not (assuming my license is assigned to the stable channel)?  This note reads like I can if that licenses statement is true early in the note, but then later in the note it reads like I always have to pass the channel name now, even if I'm on stable with a license assigned to stable

paigecalvert commented:
Came here to ask the same question as @AmberAlston 🙂 

This statement makes it sound like omitting the channel slug no longer defaults to Stable:
>Previously, omitting the channel slug defaulted to the `stable` channel.

These statements make it sound like omitting the channel slug _does_ still default to stable, and for that reason I can omit the channel slug from the install command for stable licenses:
>...and the license does not include access to the `stable` channel.
>
>For instance, `kots install app` will fail if the license only permits access to the `beta` channel, requiring the installation command to be `kots install app/beta` instead.

My other thought was that we could classify this as a breaking change, since it sounds like it could potentially break any existing automation that the vendor has set up. (Probably should either fall under a known issue or a breaking change, and makes sense why you'd say it's not a known issue)

brianrhea commented:
@paigecalvert @AmberAlston Thanks for that feedback, I've updated it to reflect that the install command still defaults to stable if there isn't a slug and added a note that `kots install app` will still work if the license has access to `stable`

ajp-io commented on docs/release-notes/rn-app-manager.md:
```diff
@@ -38,6 +38,20 @@ Support for Kubernetes: 1.28, 1.29, and 1.30
+
+However, if the license does have access to the `stable` channel, `kots install app` will proceed without errors.
+
+When the channel slug is omitted, KOTS defaults to the `stable` channel and will attempt to pull application details from that channel. Prior to `1.113.0`, this was permitted which could potentially cause mismatches in deployment settings such as logos and minimal RBAC configurations.
```
https://docs.replicated.com/reference/custom-resource-application#icon
```suggestion
When the channel slug is omitted, KOTS defaults to the `stable` channel and will attempt to pull application details from that channel. Prior to `1.113.0`, this was permitted which could potentially cause mismatches in deployment settings such as icons and minimal RBAC configurations.
```

ajp-io replied:
The second sentence is unclear. I think you're trying to say that what was previously permitted was a kots install where the channel wasn't specified but the license didn't have access to stable. But here it sounds like you're saying the behavior of defaulting to stable was previously permitted, even though that's still how it works today

ajp-io commented on docs/release-notes/rn-app-manager.md:
```diff
@@ -38,6 +38,20 @@ Support for Kubernetes: 1.28, 1.29, and 1.30
+
+**Resolution:** If the installation is for a channel other than `stable`, ensure the channel slug in the `kots install` command is explicitly specified and matches the license's assigned channel. Refer to the Vendor Portal installation instructions or use the `replicated channel inspect CHANNEL_ID` command in the Replicated CLI for the correct commands.
+
+**Alternative:** If you cannot update your command immediately, temporarily revert to KOTS version 112.
```
```suggestion
**Alternative:** If you cannot update your command immediately, temporarily revert to KOTS 1.112.4 or earlier.
```

paigecalvert commented:
@brianrhea I did a pass to reorg this a bit so that there wasn't too large of a red block at the top of the release. Instead, there's the small warning at the top that links down to the Breaking Change section for more info. I added the same message to the release note for multi-channel licenses under New Features that also points to the breaking change. 

I also did some edits to the wording to clarify what was permitted before, and why we not longer want to permit that thing (could use a review)

"Breaking Change" felt right because this change can cause an error. Another option might be to put this info under a heading like "Multi-Channel License Installation Error", "Installation Requirement for Multi-Channel Licenses", "Updating Automation and Documentation to Support Mutli-Channel Licenses"...

brianrhea commented:
@paigecalvert I like the formatting updates! However, I believe some of those revisions are inaccurate.

> when the channel slug is omitted from the kots install command (for example, kots install app), KOTS defaults to installing the latest release from the Stable channel

> Previously, any license regardless of its assigned channel could install from the Stable channel by excluding the channel slug from the kots install command.

If a license was provided, that's the channel the app comes from. The thing it's defaulting to pulling from `stable` is just the metadata like icons and RBAC config.

But @ajp-io can confirm this and we can revise back from there if confirmed

ajp-io commented:
Yes, the app always came from the channel the license was assigned to. But the channel slug in the `kots install` command is used to get things that are needed for KOTS to be set up before the app (namely, branding, minimal RBAC setting, etc.).

So previously you could install KOTS with metadata from the stable channel, but then it would be switched to channel from the license when the license was uploaded.

paigecalvert commented:
@brianrhea ok thank you! Let me know if that latest edit looks right.

paigecalvert commented on docs/release-notes/rn-app-manager.md:
```diff
@@ -38,16 +38,34 @@ Support for Kubernetes: 1.28, 1.29, and 1.30
 
+### Breaking Change {#breaking-changes-1-113-0}
+
+In KOTS 1.113.0 and later, the following error will occur during installation if the `kots install` command lacks a channel slug _and_ the license does not have access to the Stable channel: `"failed to verify and update license: requested channel not found in latest license"`. This can break any existing automation that includes a `kots install` command without a channel slug.
```
Do you mean that it could break an existing installation if there's no intervention? Or just that you'd have to be sure to start using the channel slug for future installs (maybe "breaking the docs")? 

This also makes me wonder about upgrades.

ajp-io replied:
Future installs. Like if they documented it without the channel slug. Upgrades should be fine because you don't give the admin-console upgrade command a channel slug AFAIK.

paigecalvert replied:
Updated!
